xpra/server/proxy/proxy_server.py

```diff
@@ -390,7 +390,7 @@ def disconnect(reason, *extras):
         sns = typedict(c.dictget("start-new-session", {}))
         authlog("proxy_session: displays=%s, start_sessions=%s, start-new-session=%s",
                 displays, self._start_sessions, sns)
-        if not displays or sns:
+        if not displays or (sns and len(sns) > 0):
```

Comment:
I don't understand what this is meant to change.
There is no need to test the length. An empty dictionary (or its subclass `typedict`) evaluates to `False`:
```
>>> bool(typedict())
False
>>> bool(typedict({}))
False
>>> bool(typedict({"mode" : "start"}))
True
```